src/rust/engine/fs/store/src/lib.rs

```diff
@@ -385,7 +385,11 @@ impl Store {
     let mut directories = Vec::new();
     tree.walk(&mut |_, entry| match entry {
       directory::Entry::Directory(d) => {
-        directories.push((Some(d.digest()), d.as_remexec_directory().to_bytes()))
+        let directory = d.as_remexec_directory();
+        if cfg!(debug_assertions) {
+          protos::verify_directory_canonical(d.digest(), &directory).unwrap();
```

Comment:
Maybe use `expect` with a message to better identify what this assertion is for?

Reply:
`expect` results in less information, since it discards the actual error... but might be a bit better with `unwrap_or_else(|e| panic!("... {e}")).

Reply:
Actually... in the case where we panic, the extra context isn't necessary, because we'll have a rust backtrace.

Given that, I'll land as is.